api/src/utils/parse-filter-key.ts

```diff
@@ -0,0 +1,18 @@
+/**
+ * Strip the function declarations from a list of fields
+ */
+export function parseFilterKey(key: string) {
+	const match = key.match(/^([^()]+)(\(([^)]+)\))?/);
+
+	const fieldNameWithFunction = match?.[3]?.trim();
+	const fieldName = fieldNameWithFunction || key.trim();
+	let functionName;
+
+	if (fieldNameWithFunction) {
+		functionName = match?.[1]?.trim();
+
+		return { fieldName, functionName };
+	}
+
+	return { fieldName, functionName };
+}
```

Comment:
This might be a bit premature optimization, but what about moving the regexp outside of the function so it doesn't need to be created and gc'ed every function call and exiting the function early?

```suggestion
const FILTER_KEY_REGEX = /^([^()]+)(\(([^)]+)\))?/;

/**
 * Strip the function declarations from a list of fields
 */
export function parseFilterKey(key: string) {
	const match = key.match(FILTER_KEY_REGEX);

	const fieldNameWithFunction = match?.[3]?.trim();
	const fieldName = fieldNameWithFunction || key.trim();

	if (!fieldNameWithFunction) return { fieldName };

	return { fieldName, functionName: match?.[1]?.trim() };
}
```

Reply:
Yes ❤️🚀